core/src/main/java/tech/tablesaw/columns/numbers/DoubleDataWrapper.java

```diff
@@ -0,0 +1,158 @@
+
+import it.unimi.dsi.fastutil.doubles.DoubleArrayList;
+import it.unimi.dsi.fastutil.doubles.DoubleArrays;
+import it.unimi.dsi.fastutil.doubles.DoubleComparator;
+import it.unimi.dsi.fastutil.doubles.DoubleList;
+
+import java.util.Arrays;
+import java.util.Iterator;
+
+import static tech.tablesaw.api.NumberColumn.MISSING_VALUE;
+
+public class DoubleDataWrapper implements NumericDataWrapper {
+
+    /**
+     * Compares two doubles, such that a sort based on this comparator would sort in descending order
+     */
+    private final DoubleComparator descendingComparator = (o2, o1) -> (Double.compare(o1, o2));
+
+    private DoubleArrayList data;
+
+    public DoubleDataWrapper(DoubleArrayList data) {
+        this.data = data;
+    }
+
+    @Override
+    public NumberIterator numberIterator() {
+        return new NumberIterator(data);
+    }
+
+    @Override
+    public Iterator<Double> iterator() {
+        return data.iterator();
+    }
+
+    @Override
+    public NumericDataWrapper top(int n) {
+        final DoubleArrayList top = new DoubleArrayList();
+        final double[] values = data.toDoubleArray();
+        DoubleArrays.parallelQuickSort(values, descendingComparator);
+        for (int i = 0; i < n && i < values.length; i++) {
+            top.add(values[i]);
+        }
+        return new DoubleDataWrapper(top);
+    }
+
+    @Override
+    public NumericDataWrapper bottom(int n) {
+
+        final DoubleArrayList bottom = new DoubleArrayList();
+        final double[] values = data.toDoubleArray();
+        DoubleArrays.parallelQuickSort(values);
+        for (int i = 0; i < n && i < values.length; i++) {
+            bottom.add(values[i]);
+        }
+        return new DoubleDataWrapper(bottom);
+    }
+
+    @Override
+    public int size() {
+        return data.size();
+    }
+
+    public void append(final float f) {
+        data.add(f);
+    }
+
+    @Override
+    public void append(double d) {
+        data.add(d);
+    }
+
+    @Override
+    public void append(int i) {
+        data.add(i);
+    }
+
+    @Override
+    public double getDouble(int row) {
+        return data.getDouble(row);
+    }
+
+    @Override
+    public DoubleDataWrapper copy() {
+        return new DoubleDataWrapper(data.clone());
+    }
+
+    @Override
+    public void clear() {
+        this.data.clear();
+    }
+
+    @Override
+    public void sortAscending() {
+        Arrays.parallelSort(data.elements());
+    }
+
+    @Override
+    public void sortDescending() {
+        DoubleArrays.parallelQuickSort(data.elements(), descendingComparator);
+    }
+
+    @Override
+    public void set(int r, double value) {
+        this.data.set(r, value);
+    }
+
+    @Override
+    public boolean contains(double value) {
+        return data.contains(value);
+    }
+
+    @Override
+    public boolean contains(int value) {
+        return data.contains(value);
+    }
+
+    @Override
+    public DoubleList dataInternal() {
```

Comment:
maybe we should have `intsInternal()` and `doublesInternal()`? I think throwing an `UnsupportedOperationException` would be preferable to returning null because it will be easier for the user to understand what's happening

Reply:
i'm removing this method